src/fork_gc.c

```diff
@@ -1181,6 +1196,67 @@ static FGCError FGC_parentHandleMissingDocs(ForkGC *gc) {
   return status;
 }
 
+static FGCError FGC_parentHandleExistingDocs(ForkGC *gc) {
+  FGCError status = FGC_COLLECTED;
+
+  size_t ei_len;
+  char *empty_indicator = NULL;
+
+  if (FGC_recvBuffer(gc, (void **)&empty_indicator, &ei_len) != REDISMODULE_OK) {
+    return FGC_CHILD_ERROR;
+  }
+
+  if (empty_indicator == RECV_BUFFER_EMPTY) {
+    return FGC_DONE;
+  }
+
+  InvIdxBuffers idxbufs = {0};
+  MSG_IndexInfo info = {0};
+  if (FGC_recvInvIdx(gc, &idxbufs, &info) != REDISMODULE_OK) {
+    rm_free(empty_indicator);
+    return FGC_CHILD_ERROR;
+  }
+
+  StrongRef spec_ref = WeakRef_Promote(gc->index);
+  IndexSpec *sp = StrongRef_Get(spec_ref);
+  if (!sp) {
+    status = FGC_SPEC_DELETED;
+    goto cleanup;
+  }
+
+  RedisSearchCtx sctx_ = SEARCH_CTX_STATIC(gc->ctx, sp);
+  RedisSearchCtx *sctx = &sctx_;
+
+  RedisSearchCtx_LockSpecWrite(sctx);
+
+  InvertedIndex *idx = sp->existingDocs;
+
+  FGC_applyInvertedIndex(gc, &idxbufs, &info, idx);
+  // We don't count the records that we removed, because we also don't count
+  // their addition (they are duplications so we have no such desire).
```

Comment:
Still don't we want to have some stats for the `existingDocs` ?

Reply:
We can add some, but I don't think we want to use the existing ones as it may cause more confusion than clarity.
Our report of the inverted index size includes the `existingDocs` inverted index